service/k8sclient/internal/kubectl/kubectl.go

```diff
@@ -283,3 +332,130 @@ func run(ctx context.Context, kubectlCmd []string, args []string, stdin interfac
 	l.Debug(errBuf.String())
 	return outBuf.Bytes(), err
 }
+
+var (
+	reservedProxyPortsMutex *sync.Mutex = new(sync.Mutex)
+	reservedProxyPorts      *sync.Map   = new(sync.Map)
+)
+
+func (k *KubeCtl) reserveProxyPort(port string) {
+	mutex, _ := reservedProxyPorts.LoadOrStore(port, new(sync.Mutex))
+	mutex.(*sync.Mutex).Lock()
+}
+
+func (k *KubeCtl) releaseProxyPort(port string) {
+	mutex, ok := reservedProxyPorts.Load(port)
+	if !ok {
+		return
+	}
+	mutex.(*sync.Mutex).Unlock()
+}
+
+// RunProxy runs kubectl proxy on port that is returned.
+func (k *KubeCtl) RunProxy(ctx context.Context) (string, error) {
+	k.currentProxyMutex.Lock()
+	defer k.currentProxyMutex.Unlock()
+
+	if k.currentProxy != nil {
+		return "", errors.New("kubectl proxy is already running")
+	}
+
+	var port string
+	// Try to run kubectl proxy on random port.
+	err := retry.Do(
+		func() error {
+			// Generate a random port.
```

Comment:
```suggestion
			// Generate a random port number between 10000 and 19999.
```

Reply:
changed the reserveProxyPort, it includes the comment.